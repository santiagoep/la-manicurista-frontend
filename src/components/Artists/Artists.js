import { useMemo } from 'react';
import PropTypes from 'prop-types';

import Artist from '../Artist/Artist';
import StyledContainer from './Artists.styled';

const Artists = ({ artists }) => {
  const memoizedArtists = useMemo(
    () =>
      artists.map((artist, index) => (
        <Artist
          {...artist}
          key={`artist-${artist.name}-${index}`}
          className='artists__artist'
        />
      )),
    [artists]
  );
  return <StyledContainer>{memoizedArtists}</StyledContainer>;
};

Artists.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape(Artist.propTypes))
};

Artists.defaultProps = {
  artists: []
};

export default Artists;
