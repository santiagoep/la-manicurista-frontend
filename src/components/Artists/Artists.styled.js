import styled from 'styled-components';

export default styled.div`
  display: flex;
  flex-wrap: wrap;
  > .artists__artist:after {
    content: ' &';
  }
  > .artists__artist:last-of-type:after {
    content: '';
  }
`;
