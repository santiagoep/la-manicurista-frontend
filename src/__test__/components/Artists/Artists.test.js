import React from 'react';

import render from '../../render';
import Artists from '@components/Artists/Artists';
import artistsMock from '@__mocks__/components/artists.mock';

describe('<Artists />', () => {
  it('should render snapshot', () => {
    const { renderer } = render({ theme: true, redux: true });
    const component = renderer(<Artists artists={artistsMock} />);
    expect(component).toMatchSnapshot();
  });
});
