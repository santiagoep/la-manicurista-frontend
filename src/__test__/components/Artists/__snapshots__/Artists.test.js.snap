// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Artists /> should render snapshot 1`] = `
<div
  className="Artistsstyled-sc-1qoofk0-0 iUjAkI"
>
  <p
    className="Artiststyled-sc-19exrv8-0 bkPnvl artists__artist"
  >
    Artists 1
  </p>
  <p
    className="Artiststyled-sc-19exrv8-0 bkPnvl artists__artist"
  >
    Artists 2
  </p>
</div>
`;
